src/lib/Form/EventListener/AddSiteAccessFieldBasedOnContentListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\EventListener;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use EzSystems\EzPlatformAdminUi\Form\Data\Content\CustomUrl\CustomUrlAddData;
+use EzSystems\EzPlatformAdminUi\Siteaccess\NonAdminSiteaccessResolver;
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormEvent;
+
+class AddSiteAccessFieldBasedOnContentListener
```

Comment:
I'm not sure if we expect this class to be extended? It contains `protected` method, but is there any user use case that warrants this class not being final?

Reply:
The rest of the classes are also expected to be extended in this form type. Just wanted to keep consistent, however, I've changed it to the final.